sda/simulation.go

```diff
@@ -106,35 +106,24 @@ func LoadSimulationConfig(dir, ha string) ([]*SimulationConfig, error) {
 			ha += ":"
 		}
 		for _, e := range sc.Roster.List {
-			for _, a := range e.Addresses {
-				log.Lvl4("Searching for", ha, "in", a)
-				// If we are started in Deterlab- or other
-				// big-server-needs-multiple-hosts, we might
-				// want to initialise all hosts in one instance
-				// of 'cothority' so as to minimize memory
-				// footprint
-				if strings.Contains(a, ha) {
-					log.Lvl3("Found host", a, "to match", ha)
-					host := NewHost(e, scf.PrivateKeys[a])
-					scNew := *sc
-					scNew.Host = host
-					scNew.Overlay = host.overlay
-					ret = append(ret, &scNew)
-				}
-
-			}
+			host := NewHost(e, scf.PrivateKeys[e.Address])
+			scNew := *sc
+			scNew.Host = host
+			scNew.Overlay = host.overlay
+			ret = append(ret, &scNew)
 		}
 		if len(ret) == 0 {
 			return nil, errors.New("Didn't find address: " + ha)
 		}
 	} else {
 		ret = append(ret, sc)
 	}
-	if strings.Contains(sc.Roster.List[0].Addresses[0], "localhost") {
+	addr := string(sc.Roster.List[0].Address)
+	if strings.Contains(addr, "127.0.0.") || strings.Contains(addr, "localhost") {
```

Comment:
Again - does this still work? The idea here is that for localhost-simulations, the `Conode`s are also split into different `cothorityd`-processes, and that they have a different `127.0.0.n` address.